cypress.config.ts

```diff
@@ -0,0 +1,56 @@
+import { defineConfig } from "cypress";
+import webpackConfig from "./cypress/webpack.config.js";
+
+async function setupNodeEvents(
+  on: Cypress.PluginEvents,
+  config: Cypress.PluginConfigOptions
+): Promise<Cypress.PluginConfigOptions> {
+  on("task", {
+    log(message) {
+      console.log(message);
+      return null;
+    },
+
+    table(message) {
+      console.table(message);
+      return null;
+    },
+  });
+
+  return config;
+}
+
+export default defineConfig({
+  blockHosts: ["www.google-analytics.com", "sockjs*", "countries*"],
+  env: {
+    TAGS: "not @ignore",
+    iframe: "iframe.html?id=",
+  },
+  video: false,
+  viewportWidth: 1366,
+  viewportHeight: 768,
+  chromeWebSecurity: false,
+  projectId: "8458bb",
+  retries: {
+    runMode: 1,
+    openMode: 1,
+  },
+  e2e: {
+    setupNodeEvents,
+    baseUrl: "http://127.0.0.1:9001/",
+    specPattern: "./cypress/e2e/**/*.test.ts",
+  },
+  component: {
+    devServer: {
+      framework: "react",
+      bundler: "webpack",
+      webpackConfig,
+    },
+    watchForFileChanges: true,
+    excludeSpecPattern: ["**/examples/*", "**/*.spec.{js|ts*}"],
```

Comment:
Why is this line needed? Does the following `specPattern` line not just confirm which file types are required to run?

Reply:
we are excluding unit tests :)

Reply:
So you have to state specifically what you want to include and not include? I thought that, in the previous version, stating the `specPattern` meant that only those patterns would be included and all others would be excluded. Is that not the case?